### DISCUSSION TITLE
How I "solved" troubles paring TRV SEA802-Z01 (white LED display on the side, rotary plate) and Slae.sh

### DISCUSSION THREAD

**Discussion by hnykda** [Koenkk/zigbee2mqtt#10774] [Devices]
Hey.

I had problems pairing [TRV SEA802-Z01](https://www.zigbee2mqtt.io/devices/SEA801-Zigbee_SEA802-Zigbee.html) for many months. Today I decide to either somehow make them work or just throw them away. And I somehow managed to connect them all! This post is for everyone who's struggling (and for me to remember that I did this) - that you maybe just need to keep trying + bring it close to the coordinator.

I knew they worked fine with my old CC2531, but I had to repair all devices as I bought Slae.sh (you don't have to with new versions, but I had to for some other reasons) and I didn't manage to pair these for some weird reason. I put them in the pairing mode as before by turning them first off via the rotary plate (until it shows `OF`) and then pressing the plate until `--` shows followed by `PA` characters. That all showed up, but they didn't connect to my zigbee2mqtt (that was in the "Allowed to join" mode).

What I did was the following:
1. bringing them as close as possible to the coordinator
2. taking the batteries off, then putting them in
3. waiting until it calibrates (shows `LA`)
4. then putting them in pairing mode (while setting z2m to "Allow new devices to join"), **while holding the button for as short as possible, meaning that I released the push of the rotatory plate as soon as I saw `--` on the display**

and repeated the process about ~20 times until it once succeeded :-D . Once it went through, it worked again and again (I tried to remove it again) with the same TRV, don't have idea why. I also sometimes just tried to press the button e.g. every 100ms after putting them into the pairing mode just to somehow "keep it alive", but that's probably just useless placebo, nevertheless, it connected. 

I suspect there is something broken with pairing via routers, and once the TRV gets close to the coordinator, it connects fine.

I am also trying to figure out what `system_mode` values mean. My current take:

  - `off` - puts the device into `OF` mode, closes the valve permanently
  - `heat` - opens the valve when `local_temperature + local_temperature_calibration < current_heating_setpoint` . Imho this is what you want with HA (and control it from HA only via `current_heating_setpoint`).
  - `auto` - enables automatic program and then behaves as `heat`? Dunno, it doesn't have time settings at all :thinking: . When I put it to auto, it seems that it jump to 16°C sometimes :shrug: .

**Comment by Gadzhi13** [comment]
Thanks for the help! Been struggling for the last couple of days trying to connect mine.

Just in case someone else is also struggling - my solution was a tiny bit different - I had to unplug all routers (in my case lightbulbs) in the room and then it all worked fine. Bringing the Thermostat closer to the coordinator did not help at all.
